client/extensions/woocommerce/app/products/product-create.js

```diff
@@ -127,7 +123,9 @@ class ProductCreate extends React.Component {
 
 function mapStateToProps( state ) {
 	const site = getSelectedSiteWithFallback( state );
-	const product = getCurrentlyEditingProduct( state );
+	const productId = getCurrentlyEditingId( state, site.id );
```

Comment:
I guess I was getting too ambitious about not relying on those defaults. I dialed it back for now.